spec/controllers/news_controller_spec.rb

```diff
@@ -30,15 +30,34 @@
 
       expect(assigns(:news_article)).to eq(news_article)
     end
+  end
 
+  describe 'GET #show' do
     context 'when a news article does not exist' do
+      before do
+        allow_any_instance_of(Core::NewsArticleReader).to receive(:call).and_yield(Core::NewsArticle.new('do-not-exist'))
+      end
 
       it 'raises an ActionController RoutingError' do
-        allow(Core::NewsArticleReader).to receive(:new) { ->(&block) { block.call } }
-
         expect { get :show, id: 'foo', locale: I18n.locale }.to raise_error(ActionController::RoutingError)
       end
     end
+
+    context 'when news article is redirected' do
+      before do
+        allow_any_instance_of(Core::NewsArticleReader).to receive(:call).and_yield(OpenStruct.new(status: 301, location: 'https://example.com', redirect?: true))
```

Comment:
Just to note, and I'm not suggesting you necessarily need to action this as part of this PR, but some of these lines are a bit long, longer than the max width of GitHub's PR diff. As GitHub's diffs don't line-wrap we're having to horizontally scroll to review the entire line.

So yes, intermediate variables or even simply breaking chained statements onto multiple lines would be a big help for review.